### ISSUE TITLE
New Domain Block Level: "Flag"

### ISSUE THREAD

**Issue by emmetoneillpdx** [mastodon/mastodon#21975]
### Pitch

I'd like to suggest a new level of domain blocking that I can only think to call "flagging", in which the admins of a server are able to "flag" another server as potentially problematic or offensive to some of their users. 

From a user standpoint this would mean a few things:

- A small warning icon/flag would appear on posts from flagged instances. A tool-tip explaining why posts from this server are flagged would appear when the user hovers over the icon.
- On an individual level, users should have the option to opt-in to limiting/suspending servers that have been flagged by their moderators. In other words, users could decide for themselves (by changing a user setting) how they would like to treat the servers that have been flagged.

This might be considered a new, softer level of moderation.

### Motivation

Right now, server admins have some important and powerful tools for moderating content and protecting their community. Media rejection is necessary to keep illegal media completely off of servers, and domain-level limiting gives moderators the power to shield sensitive communities from blatantly offensive content. And while I think it's not something to be taken lightly, there is absolutely a time and place for domain-level suspension. In tight-knit communities that are mostly ideologically homogeneous and where people might share similar belief systems and moderation philosophies, these tools alone can work well.

On the other hand, In my opinion, _a major drawback to the current suite of domain-level tools is that they diminish (in the case of domain-level limiting) or completely remove (in the case of domain-level suspending) the agency of the user to decide for themselves what content they see and who they communicate with_. And in larger communities, where beliefs, politics, values, cultural sensitivities, etc., are naturally more diverse, it becomes that much more important for individual users to have agency over how their own feed is moderated.

Of course, this would be an additional level of "soft" domain-blocking that wouldn't replace or remove any of the existing tools that server admins rely on. Some communities might decided to continue relying more heavily on top-down moderation tools like limiting and suspension, while others may take a flagging-based approach that more favors individual user agency. The point here, is to _provide a range of tools that allow for both styles of moderation_, not just one.

**Comment by trwnh** [comment]
- #3890
- #11218

**Comment by emmetoneillpdx** [comment]
@trwnh related but different.

In the case of #3890, we're talking about a **user-level** feature, and in the case of #11218 we're talking about the ability to **opt out** of domain-level limits (silences). Personally I agree with both of those suggestions, too...

But the main distinction with what I'm suggesting here is creating **another level of domain-level moderation that simply gives users (a) information about potentially problematic instances and (b) the user-level tools and agency to act on that information as they see fit.** 

Thinking about it from the server admin perspective, it's another tier of moderation that lets you warn your users about domains that may be potentially problematic for your community, without resorting to the more drastic steps of silencing, or even defederating.

In other words, the _moderators inform_ the users, and the _users act_ on that information--preserving user agency.
